src/Acrobot.jl

```diff
@@ -0,0 +1,220 @@
+import Flux.Tracker
+"""
+Copyright 2013, RLPy http://acl.mit.edu/RLPy
+Credit: Alborz Geramifard, Robert H. Klein, Christoph Dann,
+               William Dabney, Jonathan P. How
+License: "BSD 3-Clause"
+Author: "Christoph Dann <cdann@cdann.de>"
+"""
+mutable struct AcrobotEnv
+    viewer
+    observation_space::Box
+    action_space::Discrete
+    state
+    dt::Float32
+    LINK_LENGTH_1::Float32  # [m]
+    LINK_LENGTH_2::Float32  # [m]
+    LINK_MASS_1::Float32  # [kg] mass of link 1
+    LINK_MASS_2::Float32  # [kg] mass of link 2
+    LINK_COM_POS_1::Float32  # [m] position of the center of mass of link 1
+    LINK_COM_POS_2::Float32  # [m] position of the center of mass of link 2
+    LINK_MOI::Float32  # moments of intertia for both links
+    MAX_VEL_1::Float32
+    MAX_VEL_2::Float32
+    AVAIL_TORQUE::Array{Float32, N} where N
+    torque_noise_max::Float32
+    book_or_nips::String
+end
+
+
+function AcrobotEnv()
+    dt = 2f-1
+    LINK_LENGTH_1 = 1f0
+    LINK_LENGTH_2 = 1f0
+    LINK_MASS_1 = 1f0
+    LINK_MASS_2 = 1f0
+    LINK_COM_POS_1 = 5f-1
+    LINK_COM_POS_2 = 5f-1
+    LINK_MOI = 1f0
+
+    MAX_VEL_1 = Float32(4 * pi)
+    MAX_VEL_2 = Float32(9 * pi)
+
+    AVAIL_TORQUE = [-1f0, 0f0, 1f0]
+    torque_noise_max = 0f0
+
+    # use dynamics equations from the nips paper or the book
+    book_or_nips = "book"
+    # action_arrow = nothing
+    # domain_fig = nothing
+    # actions_num = 3
+
+    viewer = nothing
+    high = [1f0, 1f0, 1f0, 1f0, MAX_VEL_1, MAX_VEL_2] #//
+    low = -high
+    observation_space = Box(low, high, Float32)
+    action_space = Discrete(3)
+    state = nothing
+
+    AcrobotEnv(viewer, observation_space, action_space, state, dt, LINK_LENGTH_1,
+               LINK_LENGTH_2, LINK_MASS_1, LINK_MASS_2, LINK_COM_POS_1, LINK_COM_POS_2,
+               LINK_MOI, MAX_VEL_1, MAX_VEL_2, AVAIL_TORQUE, torque_noise_max, book_or_nips)
+end
+
+
+function reset!(env::AcrobotEnv)
+    env.state = param(rand(Float32, 4) * 2f-1 .- 1f-1)
+    return _get_ob(env)
+end
+
+#=
+function custom_reset!(env::AcrobotEnv, args=[-0.04616826f0, -0.04653378f0,  0.07321809f0, -0.04461966f0])
+    env.state = param(args)
+    return _get_ob(env)
+end
+=#
+
+function step!(env::AcrobotEnv, action)
+    s = env.state
+    torque = env.AVAIL_TORQUE[action]
+
+    # Add noise to the force action
+    env.torque_noise_max > 0 && (torque += param(env.torque_noise_max * (2rand(Float32) - 1)))
```

Comment:
let's have the output of above mentioned function a `param` so that we need not noise as `param`

Reply:
@tejank10 On closer inspection, I think maybe we don't need `params` here at all. I believe I called `param` against torque in my naivety. Tracking `torque` doesn't really serve any purpose here, since it's `state` in which we are really interested. So, maybe drop `param` altogether?